tox.ini

```
@@ -106,3 +106,12 @@ depends =
 
 [coverage:run]
 relative_files = True
+
+[testenv:pack-{22.04,24.04}]
+description = Copy the corresponding charmcraft-{}.yaml to charmcraft.yaml and pack
+allowlist_externals =
+    sh
+commands =
+    22.04: sh -c "cp {toxinidir}/charmcraft-22.04.yaml {toxinidir}/charmcraft.yaml && charmcraft pack --project-dir={toxinidir}"
+    24.04: sh -c "cp {toxinidir}/charmcraft-24.04.yaml {toxinidir}/charmcraft.yaml && charmcraft pack --project-dir={toxinidir}"
```

Can you first `cp` the current `charmcraft.yaml` to `charmcraft.yaml.bak`? So we don't accidentally commit the wrong `charmcraft.yaml`.

```suggestion
    22.04: sh -c "cp {toxinidir}/charmcraft.yaml {toxinidir}/charmcraft.yaml.bak; cp {toxinidir}/charmcraft-22.04.yaml {toxinidir}/charmcraft.yaml && charmcraft pack --project-dir={toxinidir}"; mv {toxinidir}/charmcraft.yaml.bak {toxinidir}/charmcraft.yaml
    24.04: sh -c "cp {toxinidir}/charmcraft.yaml {toxinidir}/charmcraft.yaml.bak; cp {toxinidir}/charmcraft-24.04.yaml {toxinidir}/charmcraft.yaml && charmcraft pack --project-dir={toxinidir}"; mv {toxinidir}/charmcraft.yaml.bak {toxinidir}/charmcraft.yaml
```

---

Good point. Switched to using a symlink, with analogous commands.
With a relative symlink we will have only 2 copies instead of 3.
Wdyt?

---

That works!